x-pack/plugins/endpoint/server/routes/alerts.test.ts

```diff
@@ -0,0 +1,191 @@
+    expect(alertResultList.total).toEqual(132);
+    expect(alertResultList.request_page_index).toEqual(0);
+    expect(alertResultList.request_page_size).toEqual(10);
+  });
+
+  it('should return the latest of all alerts', async () => {
+    const mockRequest = httpServerMock.createKibanaRequest({});
+
+    const response: SearchResponse<AlertData> = (data as unknown) as SearchResponse<AlertData>;
+    mockScopedClient.callAsCurrentUser.mockImplementationOnce(() => Promise.resolve(response));
+    [routeConfig, routeHandler] = routerMock.post.mock.calls.find(([{ path }]) =>
+      path.startsWith('/api/endpoint/alerts')
+    )!;
+
+    await routeHandler(
+      ({
+        core: {
+          elasticsearch: {
+            dataClient: mockScopedClient,
+          },
+        },
+      } as unknown) as RequestHandlerContext,
+      mockRequest,
+      mockResponse
+    );
+
+    expect(mockScopedClient.callAsCurrentUser).toBeCalled();
+    expect(routeConfig.options).toEqual({ authRequired: true });
+    expect(mockResponse.ok).toBeCalled();
+    const alertResultList = mockResponse.ok.mock.calls[0][0]?.body as AlertResultList;
+    expect(alertResultList.total).toEqual(132);
+    expect(alertResultList.request_page_index).toEqual(0);
+    expect(alertResultList.request_page_size).toEqual(10);
+  });
+
+  it('should return alert results according to pagination params -- POST', async () => {
+    const mockRequest = httpServerMock.createKibanaRequest({
+      method: 'post',
+      body: {
+        paging_properties: [
+          {
+            page_size: 20,
+          },
+          {
+            page_index: 2,
+          },
+        ],
+      },
+    });
+    mockScopedClient.callAsCurrentUser.mockImplementationOnce(() =>
+      Promise.resolve((data as unknown) as SearchResponse<AlertData>)
+    );
+    [routeConfig, routeHandler] = routerMock.post.mock.calls.find(([{ path }]) =>
+      path.startsWith('/api/endpoint/alerts')
+    )!;
+
+    await routeHandler(
+      ({
+        core: {
+          elasticsearch: {
+            dataClient: mockScopedClient,
+          },
+        },
+      } as unknown) as RequestHandlerContext,
+      mockRequest,
+      mockResponse
+    );
+
+    expect(mockScopedClient.callAsCurrentUser).toBeCalled();
+    expect(routeConfig.options).toEqual({ authRequired: true });
+    expect(mockResponse.ok).toBeCalled();
+    const alertResultList = mockResponse.ok.mock.calls[0][0]?.body as AlertResultList;
+    expect(alertResultList.alerts.length).toEqual(20);
+    expect(alertResultList.total).toEqual(132);
+    expect(alertResultList.request_page_index).toEqual(40);
+    expect(alertResultList.request_page_size).toEqual(20);
+  });
+
+  it('should return alert results according to pagination params -- GET', async () => {
+    const mockRequest = httpServerMock.createKibanaRequest({
+      path: '/api/endpoint/alerts?page_size=20&page_index=2',
+      query: {
+        page_size: 20,
```

Comment:
Is there a reason this `page_size` value is here and in the `path` above?

Reply:
Ummm nope, just an oversight. I'll fix, thanks.